openedx/core/djangoapps/safe_sessions/tests/test_middleware.py

```diff
@@ -422,6 +428,32 @@ def test_warn_without_verbose_logging(self):
         with self.assert_regex_not_logged('SafeCookieData: Changing request user. ', log_level='warning'):
             SafeSessionMiddleware().process_response(self.request, self.client.response)
 
+    @patch("openedx.core.djangoapps.safe_sessions.middleware.LOG_REQUEST_USER_CHANGES", True)
+    @patch("openedx.core.djangoapps.safe_sessions.middleware.LOG_REQUEST_USER_CHANGE_HEADERS", True)
+    @patch("openedx.core.djangoapps.safe_sessions.middleware.cache")
+    @patch("openedx.core.djangoapps.safe_sessions.middleware.log")
+    def test_with_header_logging(self, mock_log, mock_cache):
+        self.set_up_for_success()
+        self.request.user = UserFactory.create()
+        with self.assert_logged('SafeCookieData: Changing request user. ', log_level='warning'):
+            SafeSessionMiddleware().process_response(self.request, self.client.response)
+        # Note: Since the test cache is not retaining its values for some reason, we'll
```

Comment:
**Reviewer Note:** I do not know why the test cache is not working as I'd expect. Using a breakpoint in the code, I did a set and immediate get and it was not working. I'm going to assume it is a test issue (unless this doesn't work in Stage), and just err on getting this out there sooner. Please mark resolved after reading, or let me know if you have ideas.